common/ferrostar/src/models.rs

```diff
@@ -322,6 +322,8 @@ pub struct RouteStep {
     pub spoken_instructions: Vec<SpokenInstruction>,
     /// A list of json encoded strings representing annotations between each coordinate along the step.
     pub annotations: Option<Vec<String>>,
+    /// A list of incidents that occur along the step.
+    pub incidents: Vec<Incident>,
```

Comment:
update - i just realized, that's what i am doing here and i just didn't realize it - but my logic is wrong in common/ferrostar/src/routing_adapters/osrm/mod.rs, since i need to also slice these just like we're doing with annotations. will mark this as a draft while i fix it.